### PR TITLE
fix: Phone Input and Input widget not getting reset on submit issue

### LINKED ISSUE #10859: [Bug]: reset on submit is not working for input and phone input widgets on enter
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

 reset on submit is not working for input and phone input widgets on enter

### Steps To Reproduce

1. Add a Phone input widget.
2. Enable reset on submit.
2. Add an on submit expression.
3. Type some text on input and press enter.
4. Text is not getting cleared 

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by sbalaji1192**
Fixes #10859

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/phone-input-widget-reset 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.99 **(0)** | 36.33 **(0.02)** | 34.68 **(0)** | 55.37 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :red_circle: | app/client/src/widgets/BaseInputWidget/widget/index.tsx | 65 **(-5.27)** | 6.25 **(-1.44)** | 66.67 **(0)** | 68.42 **(-5.87)**
 :green_circle: | app/client/src/widgets/CurrencyInputWidget/widget/index.tsx | 46.73 **(3.25)** | 20 **(3.93)** | 40 **(2.5)** | 46.67 **(2.53)**</details>

### CODE REVIEW

**Comment by shwetha-ramesh** [comment]
@sbalaji1192 Tested this PR. DP is working fine.
Moving to Done.

File: app/client/src/widgets/BaseInputWidget/widget/index.tsx [hunk 1]
```diff
@@ -304,7 +304,7 @@ class BaseInputWidget<
     this.props.updateWidgetMetaProperty("isFocused", focusState);
   }
 
-  onSubmitSuccess(result: ExecutionResult) {
+  onSubmitSuccess = (result: ExecutionResult) => {
     if (result.success && this.props.resetOnSubmit) {
       this.props.updateWidgetMetaProperty("text", "", {
```
Comment (jsartisan) [inline app/client/src/widgets/BaseInputWidget/widget/index.tsx:309]: Can't we use `resetChildrenMetaProperty` here?

**Comment by sbalaji1192** [comment]
@shwetha-ramesh Could test this again, I have made some changes related to resetting value and isDirty.

**Review by jsartisan — APPROVED** [review]

**Comment by shwetha-ramesh** [comment]
@sbalaji1192 tested this PR, observed this issue on all 3 input widgets.

On initial drag and drop the input widget doesn't clear the input when ```Reset on submit``` is ON and ```OnSubmit``` is No action. Later you add a Onsubmit action enter the input on the widget and click on enter the input gets cleared. Now clear the ```OnSubmit``` action to No action and enter any input and observe the input on the widget gets cleared. 

https://loom.com/share/383c13e15b3c4dee87a13aaf378e83cb

**Comment by sbalaji1192** [comment]
@shwetha-ramesh Fixed your callout. please check again

**Comment by shwetha-ramesh** [comment]
@sbalaji1192 tested this PR. DP working fine.
Moving to Done.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**